crates/noirc_evaluator/src/ssa/flatten.rs

```diff
@@ -0,0 +1,394 @@
+use super::{
+    block,
+    code_gen::IRGenerator,
+    node::{self, Node, NodeEval, Operation},
+    optim,
+};
+use acvm::FieldElement;
+use arena::Index;
+use std::collections::HashMap;
+
+//Unroll the CFG
+pub fn unroll_tree(eval: &mut IRGenerator) {
+    //Calls outer_unroll() from the root node
+    let mut id = eval.first_block;
+    let mut unroll_ins: Vec<Index> = Vec::new();
+    let mut eval_map: HashMap<Index, node::NodeEval> = HashMap::new();
+    let mut from = id;
+    while let Some(next) = outer_unroll(&mut unroll_ins, &mut eval_map, id, from, eval) {
+        from = id;
+        id = next;
+    }
+}
+
+//Update the block instruction list using the eval_map
+fn eval_block(block_id: Index, eval_map: &HashMap<Index, NodeEval>, eval: &mut IRGenerator) {
+    let block = eval.get_block(block_id).unwrap();
+    for i in &block.instructions.clone() {
+        //RIA
+        if let Some(ins) = eval.get_as_mut_instruction(*i) {
+            if eval_map.contains_key(&ins.rhs) {
+                ins.rhs = eval_map[&ins.rhs].to_index().unwrap();
+            }
+            if eval_map.contains_key(&ins.lhs) {
+                ins.lhs = eval_map[&ins.lhs].to_index().unwrap();
+            }
+            if ins.operator == Operation::eq_gate {}
```

Comment:
No, I believe it is a left-over of some debug session..
The TODO simplify is because I would like to call optim::simplify(eval, ins) here but rust does not allow it and I could not find a workaround. Do you have a solution?

Reply:
I am OK to try this but since I was not successful, I would prefer that this point does not block the merge, we can see after the merge how we can call this `simplify` method.

Reply:
I don't think this should block the merge, either. Marking this as closed